src/_semaphore.ts

```
@@ -0,0 +1,276 @@
+	private ensureSemaphoreDir(): void {
+		const dir = this.getSemaphoreDir();
+		if (!existsSync(dir)) {
+			mkdirSync(dir, { recursive: true });
+		}
+	}
+
+	/**
+	 * Gets the semaphore file path for a session
+	 */
+	private getSemaphoreFilePath(sessionId: string): string {
+		return join(this.getSemaphoreDir(), `${sessionId}.json`);
```

ESLint

[error] Unsafe return of a value of type error.

(ts/no-unsafe-return)